### PR TITLE
Fix for FormatTable remote deserialization regression

### PR DESCRIPTION
**Pull request by PaulHigin**
A recent accessibility change has caused a regression in PowerShell remoting between machines with and without the change.  The change is a new `RepeatHeader` property added to the TableHeaderInfo class.  When formatting objects are returned through the remoting channel, the client expects to see `RepeatHeader` property in the serialized TableHeaderInfo object and throws when it is missing.

Fix is to allow this serialized property to be null and use 'false' as the default value for the property.

Repro:
```powershell
$s = New-PSSession -cn <oldVersionMachine>    # Create remote connection to machine running older version of PowerShell 5 or 6
Invoke-Command -Session $s - Script { Get-Process pwsh | Format-Table }

# Expected
# Formatted process information

# Result
# TerminatingError(out-lineoutput): "The value of the repeatHeader data member cannot be null."
```

## PR Context

This breaks down level system remoting compatibility, and throws a terminating error whenever formatting objects are returned through the remoting layer.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` to your commit messages if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Review by SteveL-MSFT — APPROVED** [review]

File: src/System.Management.Automation/FormatAndOutput/common/FormattingObjectsDeserializer.cs [hunk 1]
```diff
@@ -586,8 +587,10 @@ internal sealed partial class TableHeaderInfo : ShapeInfo
         internal override void Deserialize(PSObject so, FormatObjectDeserializer deserializer)
         {
             base.Deserialize(so, deserializer);
+            // The "repeatHeader" property was added later (V5,V6) and presents an incompatibility when remoting to older version PowerShell sessions.
```
Comment (RDIL) [inline src/System.Management.Automation/FormatAndOutput/common/FormattingObjectsDeserializer.cs]:
```suggestion
            // The "repeatHeader" property was added later (V5, V6) and presents an incompatibility when remoting to older version PowerShell sessions.
```

Comment (PaulHigin) [reply src/System.Management.Automation/FormatAndOutput/common/FormattingObjectsDeserializer.cs]: Will do.  Thanks.

File: src/System.Management.Automation/FormatAndOutput/common/FormattingObjectsDeserializer.cs [hunk 2]
```diff
@@ -586,8 +587,10 @@ internal sealed partial class TableHeaderInfo : ShapeInfo
         internal override void Deserialize(PSObject so, FormatObjectDeserializer deserializer)
         {
             base.Deserialize(so, deserializer);
+            // The "repeatHeader" property was added later (V5, V6) and presents an incompatibility when remoting to older version PowerShell sessions.
```
Comment (iSazonov) [inline src/System.Management.Automation/FormatAndOutput/common/FormattingObjectsDeserializer.cs]: Please add new line before the line 590 - it is CodeFactor report.

**Review by iSazonov — APPROVED** [review]
LGTM with one minor comment.

File: src/System.Management.Automation/FormatAndOutput/common/FormattingObjectsDeserializer.cs (hunk 2 above)
Comment (PaulHigin) [reply src/System.Management.Automation/FormatAndOutput/common/FormattingObjectsDeserializer.cs]: Will do.

**Review by adityapatwardhan — APPROVED** [review]

**Comment by PaulHigin** [comment]
@adityapatwardhan  Can this PR be merged?

**Comment by BadSqlPro** [comment]
Hello, so is this fixed for the PS version 5.1? Or will this be something that we will have to the Core or PS 7 to get it working? How do I patch or update my desktop so that this issue is fixed and I can continue using my scripts on PS 5.1? Thanks

**Comment by SteveL-MSFT** [comment]
This was also fixed in PS v5.1 in Win 10 1903.  There's a separate issue with ISE that's being fixed.

**Comment by BadSqlPro** [comment]
> This was also fixed in PS v5.1 in Win 10 1903. There's a separate issue with ISE that's being fixed.

Hello @SteveL-MSFT , you mentioned that the fix was provided in PS v5.1 in Win 1903 and I provided you with a proof that I am experiencing the same issue running the version it should be fixed in. Can you please point me in a direction where I can look up what update I need to apply to either the PS or Win to get this issue resolved?

**Comment by SteveL-MSFT** [comment]
@BadSqlPro looks like this is in build 18892 or newer.  You can workaround the issue by explicitly using `Format-List` or `Format-Wide`, just not `Format-Table` (which is default for many things unfortunately)

**Pull request merged**